### DISCUSSION TITLE
Defining key values in Markdown map topic references

### DISCUSSION THREAD

**Discussion by chrispy-snps** [dita-ot/dita-ot#4490] [Q&A]
Let's say I have the following two topics, each with a key-based cross-reference:

```markdown
---
filename: userguide.md
---

# My User Guide

## Introduction

For reference information, see [My Reference Manual](refman-key).
```

```markdown
---
filename: refman.md
---
# My Reference Manual

## Introduction

For usage information, see [My User Guide](userguide-key).
```

What is the correct Markdown map syntax to specify the topic file references, navtitle, *and* key definitions?

I tried this, but it crashes:

```markdown
---
$schema: urn:oasis:names:tc:dita:xsd:map.xsd
filename: map.md
---

# My Online Help

- User Guides
  - [My User Guide](userguide.md)
- Reference Manuals
  - [My Reference Manual](refman.md)
- [userguide-key]: userguide.md
- [refman-key]: refman.md
```

And really, I want a syntax that does all three in a single reference, the Markdown map equivalent of this:

```xml
<topicref href="userguide.md" keys="userguide-key" navtitle="My User Guide"/>
<topicref href="refman.md" keys="refman-key" navtitle="My Reference Manual"/>
```

I don't see this form explicitly mentioned here:

[Markdown DITA syntax](https://www.dita-ot.org/dev/reference/markdown/markdown-dita-syntax) (DITA-OT docs)

**Comment by infotexture** [comment]
@chrispy-snps It looks like you're mixing the unordered list syntax for standard topic references with the link reference syntax, which might cause problems.

Have you tried removing the list marker hyphens from the last 2 lines?

```markdown
---
$schema: urn:oasis:names:tc:dita:xsd:map.xsd
filename: map.md
---

# My Online Help

- User Guides
  - [My User Guide](userguide.md)
- Reference Manuals
  - [My Reference Manual](refman.md)

[userguide-key]: userguide.md
[refman-key]: refman.md
```

**Reply by chrispy-snps** [reply]
Hey @infotexture!

When I try that, it doesn't crash, but I get unresolved keyrefs during the transformation:

```
$ dita -i map.md -f html5 -t temp -Dclean.temp=no
File file:/mnt/c/Users/chrispy/DITA/markdown-map/refman-key not found
File file:/mnt/c/Users/chrispy/DITA/markdown-map/userguide-key not found
```

and indeed, the cross-references are broken in the HTML.

I feel like surely others must have done this before, and I must be missing something obvious.

**Reply by stefan-jung** [reply]
> I feel like surely others must have done this before, and I must be missing something obvious.

That's a very familiar feeling

**Reply by jelovirt** [reply]
Added a note to [LwDITA plug-in docs](https://github.com/jelovirt/org.lwdita/wiki/Markdown-DITA-syntax#key-definitions) that you can't define keys in topic references in Markdown. @infotexture please feel free to improve the copy.

**Comment by jelovirt** [comment]
```markdown
For reference information, see [My Reference Manual](refman-key).
```

The problem is here. This should be a [full reference link](https://spec.commonmark.org/0.31.2/#full-reference-link):

```markdown
For reference information, see [My Reference Manual][refman-key].
```

**Comment by chrispy-snps** [comment]
@infotexture, @jelovirt - thanks very much! Indeed, I was not using the proper full reference link syntax for keyrefs.

So now I have:

```markdown
---
filename: userguide.md
---

# My User Guide

## Introduction

For reference information, see [My Reference Manual][refman-key].
```

```markdown
---
filename: refman.md
---
# My Reference Manual

## Introduction

For usage information, see [My User Guide][userguide-key].
```

```markdown
---
$schema: urn:oasis:names:tc:dita:xsd:map.xsd
filename: map.md
---

# My Online Help

- User Guides
  - [My User Guide][userguide.md]
- Reference Manuals
  - [My Reference Manual][refman.md]

[userguide-key]: userguide.md
[refman-key]: refman.md
```

and the keyrefs resolve, but the map TOC shows the keydefs explicitly instead of hiding them as resource-only:

```html
<body id="my-online-help">
   <h1 class="title topictitle1">My Online Help</h1>
   <nav>
      <ul class="map">
         <li class="topicref keydef"><a href="userguide.html">My User Guide</a></li>     <!-- HERE -->
         <li class="topicref keydef"><a href="refman.html">My Reference Manual</a></li>  <!-- HERE -->
         <li class="topicref topichead">User Guides
            <ul>
               <li class="topicref"><a href="userguide.html">My User Guide</a></li>
            </ul>
         </li>
         <li class="topicref topichead">Reference Manuals
            <ul>
               <li class="topicref"><a href="refman.html">My Reference Manual</a></li>
            </ul>
         </li>
      </ul>
   </nav>
</body>
```

so I am still missing something about how to properly use keys for Markdown map topic references (or maybe I can't?).
